### PR TITLE
core(metrics): add first CPU idle lantern metric

### PR DESCRIPTION
**Pull request by patrickhulce**
I realized that we can realistically carry over all of our existing metrics in lantern world, and really FCPUI is the only one that I didn't have an impl of yet

this just reuses the graphs and coefficients of TTI combined with the existing algorithm for first interactive that does shortening window size/task clusters/etc

by reusing the graphs and coefficients of TTI we will also always give consistent answers on TTI/TTFCPUI because in the worst case, the start of the quiet window will be the last long task (which is equal to TTI estimate)

accuracy is really quite good, spearman's rho of ~0.82, which is slightly worse than TTI but better than FCP/FMP. MAPE of ~30%, which is nearly identical to TTI absolute error

### CODE REVIEW

File: lighthouse-core/gather/computed/metrics/lantern-first-cpu-idle.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+'use strict';
+
+const Node = require('../../../lib/dependency-graph/node');
+const CPUNode = require('../../../lib/dependency-graph/cpu-node'); // eslint-disable-line no-unused-vars
+const NetworkNode = require('../../../lib/dependency-graph/network-node'); // eslint-disable-line no-unused-vars
+
+const FirstInteractive = require('../first-interactive');
+const LanternConsistentlyInteractive = require('./lantern-consistently-interactive');
+
+class FirstCPUIdle extends LanternConsistentlyInteractive {
```
Comment (brendankenny) [inline lighthouse-core/gather/computed/metrics/lantern-first-cpu-idle.js]: > this just reuses the graphs and coefficients of TTI combined with the existing algorithm for first interactive

one downside to this approach is it's getting difficult to layer all three (four?) of these in my head to follow along :)

File: lighthouse-core/test/gather/computed/metrics/lantern-first-cpu-idle-test.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    const result = await artifacts.requestLanternFirstCPUIdle({trace, devtoolsLog});
+
+    assert.equal(Math.round(result.timing), 5308);
+    assert.equal(Math.round(result.optimisticEstimate.timeInMs), 2451);
+    assert.equal(Math.round(result.pessimisticEstimate.timeInMs), 2752);
+    assert.equal(result.optimisticEstimate.nodeTiming.size, 19);
+    assert.equal(result.pessimisticEstimate.nodeTiming.size, 79);
+    assert.ok(result.optimisticGraph, 'should have created optimistic graph');
+    assert.ok(result.pessimisticGraph, 'should have created pessimistic graph');
+  });
```
Comment (brendankenny) [inline lighthouse-core/test/gather/computed/metrics/lantern-first-cpu-idle-test.js]: it feels like there should be more tests here, but I don't have any special insight as to what boundaries they should be checking. Difficult traces or whatever. They can build up as we come across the difficult cases, but we don't have much infrastructure to spot those :) We should probably get some lantern smoke tests set up.

Comment (patrickhulce) [reply lighthouse-core/test/gather/computed/metrics/lantern-first-cpu-idle-test.js]: 👍 right now its just me manually running my trace-evaluations scripts, post I/O I'll try and stick a lot of my traces onto GCP and get a travis script running to make sure accuracy doesn't regress

**Comment by patrickhulce** [comment]
> whats the relative error? That's probably the better number to give here :P

sure it's ~30% which is similar to TTI

**Comment by patrickhulce** [comment]
@paulirish did the single-line if's win you over enough to insta-approve? :)
